littlepay/api/__init__.py

```diff
@@ -8,13 +9,45 @@
 TResponse = TypeVar("TResponse")
 
 
+def from_kwargs(cls, **kwargs):
+    """
+    Helper function meant to be used as a @classmethod
+    for instantiating a dataclass and allowing unexpected fields
+
+    See https://stackoverflow.com/a/55101438
+    """
+    # fetch the constructor's signature
+    class_fields = {field for field in signature(cls).parameters}
+
+    # split the kwargs into native ones and new ones
+    native_args, new_args = {}, {}
+    for name, val in kwargs.items():
+        if name in class_fields:
+            native_args[name] = val
+        else:
+            new_args[name] = val
+
+    # use the native ones to create the class ...
+    instance = cls(**native_args)
+
+    # ... and add the new ones by hand
+    for new_name, new_val in new_args.items():
+        setattr(instance, new_name, new_val)
```

Comment:
Let's actually skip this part. I'd rather we know about these fields and set them explicitly via our `@dataclass` signatures.

Reply:
Indeed, there's not much to do with unexpected args. Maybe log and monitor this particular log to detect such changes early on ?